src/components/App/ProtectedRoute.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import { Route, Redirect } from 'react-router'
+import { selectIsPendingToken, selectToken } from '../../utils/selectors'
+import { useSelector } from 'react-redux'
+import { authInstance } from '../../firebase/firebase'
+
+interface ProtectedRouteProps {
+    component: React.ElementType
+    [x: string]: any
+}
+
+const ProtectedRoute = ({
+    component: Component,
+    ...rest
+}: ProtectedRouteProps) => {
+    const isAnonymous = authInstance.currentUser?.isAnonymous
+    const token = useSelector(selectToken)
+    const isPending = useSelector(selectIsPendingToken)
+
+    if (isPending) return null
+
+    return (
+        <Route
+            {...rest}
+            render={(props) => {
+                return !isAnonymous && token ? (
+                    <Component {...props} />
```

Comment:
if you meant to create a generic `ProtectedRoute`, pls mind that an anonymous user is also a logged-in user, and needs to be able accessing `ProtectedRoutes`, the difference is:
- an anonymous user:
    - is redirected to the home page after login
    - cannot create premium video calls (but can join)
- a non-anonymous user:
    - is redirected to the *admin dashboard* after login
    - can create & join premium video calls

Reply:
You may rename the component to `PremiumRoute` maybe, and create another one for `ProtectedRoute`

Reply:
It was a little bit confusing. Thanks for the explanations. It's ok now